homeassistant/components/todoist/types.py

```diff
@@ -12,3 +13,40 @@ class DueDate(TypedDict):
     lang: str
     string: str
     timezone: str | None
+
+
+class ProjectData(TypedDict):
```

Comment:
It would be easier to review the focused API changes if the data model cleanup was also done first in a separate PR. Then it is significantly faster to review the deltas due to API changes vs cleanup from dicts to type dicts.  Can you separate this out? I'm happy to review quickly if you tag me or assign me.